client/state/country-states/reducer.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	COUNTRY_STATES_RECEIVE,
+	COUNTRY_STATES_REQUEST,
+	COUNTRY_STATES_REQUEST_FAILURE,
+} from 'state/action-types';
+import { itemSchema } from './schema';
+import { createReducer } from 'state/utils';
+
+// Stores the complete list of states, indexed by locale key
+export const items = createReducer( {}, {
+	[ COUNTRY_STATES_RECEIVE ]: ( state, action ) => ( { ...state, [ action.countryCode ]: action.countryStates } ),
+}, itemSchema );
+
+// Tracks states list fetching state
+export const isFetching = createReducer( {}, {
+	[ COUNTRY_STATES_REQUEST ]: ( state, { countryCode } ) => ( { ...state, [ countryCode ]: true } ),
+	[ COUNTRY_STATES_RECEIVE ]: ( state, { countryCode } ) => ( { ...state, [ countryCode ]: false } ),
```

Comment:
@obenland : Yeah, we're unfortunately not very consistent. There was some [discussion a while back](https://github.com/Automattic/wp-calypso/pull/1660#r47770586) about why `_REQUEST` suffix might be more preferable to `_FETCH`, though I've never felt too strongly about it (aside from avoiding inconsistencies).

There's no real difference between `_REQUEST` and `_FETCH` in most cases. `_RECEIVE` is more generic in that items are received, whether from an API request or another source. That's why, if you're concerned with tracking a request's progress, I'd recommend a separate `_REQUEST_SUCCESS` action.

Example: https://github.com/Automattic/wp-calypso/blob/3eba500/client/state/page-templates/actions.js#L43-L47